airflow/api_fastapi/core_api/datamodels/assets.py

```diff
@@ -99,3 +99,22 @@ class AssetEventCollectionResponse(BaseModel):
 
     asset_events: list[AssetEventResponse]
     total_entries: int
+
+
+class CreateAssetEventsBody(BaseModel):
+    """Create asset events request."""
+
+    uri: str
+    extra: dict = Field(default_factory=dict)
+
+    @field_validator("extra", mode="before")
+    def set_from_rest_api(cls, v):
+        if not isinstance(v, dict):
+            v = {}
+        v.setdefault("from_rest_api", True)
+        return v
```

Comment:
We shouldn't use `setdefault` because even if the payload is stating that `from_rest_api` is False, it should be overridden and set to True, because we are actually creating the event from the API. (This was the behavior in the legacy, I think we should keep it).

If v is not a dict, there is nothing we can do, we need to return `v` and let pydantic handle the rest for us.

Also if you tranform this `before` validator into an `after` validator, you will actually be able to type the function, because we know for sure that `extra` is a dict at this point. (in an after validator), also simplifying the implementation of the validator.

```suggestion
    def set_from_rest_api(cls, v):
        if isinstance(v, dict):
            v["from_rest_api"], True)
        return v        
```

Reply:
Done